src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -149,69 +148,21 @@ proc createMessageItemsFromMessageDtos(self: Module, messages: seq[MessageDto],
       if message.transactionParameters.fromAddress != "":
         isCurrentUser = self.currentUserWalletContainsAddress(message.transactionParameters.fromAddress)
 
-    var item = initMessageItem(
-      message.id,
-      message.chatId,
+    var item = message_model.createMessageItemFromDtos(
+      message,
       message.communityId,
-      message.responseTo,
-      message.`from`,
-      sender.defaultDisplayName,
-      sender.optionalName,
-      sender.icon,
-      sender.colorHash,
-      (isCurrentUser and message.contentType != ContentType.DiscordMessage),
-      sender.dto.added,
-      message.outgoingStatus,
+      sender,
+      isCurrentUser,
       renderedMessageText,
-      message.text,
-      message.parsedText,
-      message.image,
-      message.containsContactMentions(),
-      message.seen,
-      timestamp = message.timestamp,
-      clock = message.clock,
-      message.contentType,
-      message.messageType,
-      message.contactRequestState,
-      sticker = message.sticker.url,
-      message.sticker.pack,
-      message.links,
-      message.linkPreviews,
-      newTransactionParametersItem(message.transactionParameters.id,
-        message.transactionParameters.fromAddress,
-        message.transactionParameters.address,
-        transactionContract,
-        transactionValue,
-        message.transactionParameters.transactionHash,
-        message.transactionParameters.commandState,
-        message.transactionParameters.signature),
-      message.mentionedUsersPks(),
-      sender.dto.trustStatus,
-      sender.dto.ensVerified,
-      message.discordMessage,
-      resendError = "",
-      message.deleted,
-      message.deletedBy,
+      clearText = message.text,
+      albumImages = if (len(message.albumId) == 0): @[] else: @[message.image],
+      albumMessageIds = if (len(message.albumId) == 0): @[] else: @[message.id],
```

Comment:
These 2 lines with album image stuff get repeated over the whole PR. Since we already pass `message` into the `createMessageItemFromDtos` function, could we maybe simplify this as well? :)

Reply:
Done